api.ts

```diff
@@ -61,11 +62,25 @@ export class UserVisibleError extends Error {
   }
 }
 
+interface StatusCodeErrorResponse {
+  body?: any;
+  headers?: {[key: string]: string | string[] | undefined};
+}
+
+/**
+ * StatusCodeError is a simple version of StatusCodeError in request-promise to keep backwards compatibility.
+ */
 export class StatusCodeError extends Error {
   statusCode: number;
-  constructor(statusCode: number) {
-    super(`statusCode: ${statusCode}`);
+  body: any;
+  options: FetchRequest;
+  response: StatusCodeErrorResponse;
```

Comment:
Was just seeing the usage in the `packs` change--are all these fields intentionally required?

Reply:
yeah, basically because statuscodeerror (which usually returns from request-promise) is treated as a user visible error, and we need extra information for logging / display purpose. 

the fact with packs repo is that we sometimes try to create a fake status code error. they display similarly to a real network error.